### PR TITLE
Send user events to Instabug

### LINKED ISSUE #3738: Send user events to Instabug so that surveys can be triggered
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a UXR, I want to create Instabug surveys after specific app events so that I can survey users with relevant context.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: 

UXR is tasked with creating surveys to measure user feedback and OKR results such as ">95% of 20+ people surveyed trust Status for messaging".  In order to ensure the survey responses are relevant they need to be shows to users at the right time. For example, showing a messaging reliability survey to users when they first open the app won't yield meaningful answers, but showing the same survey to users after they send a message will ensure the users have the proper messaging context.

Instabug has a feature that can show surveys to users based on events and conditions, but in order for it to work events need to be sent to Instabug. Currently no events are sent.

#### Expected behavior
[comment]: # (Describe what you expected to happen.)

UXR creates a survey and sets a conditions to show the survey after a specific event.
When the app users completes the event, the survey is shown.

#### Actual behavior
[comment]: # (Describe what actually happened.)

UXR creates a survey, but can't set conditions because no events are being sent. 
Without conditions the survey is shown to all users at the same time.

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 

Status now sends (opt-in) user events to Mixpanel (see here https://github.com/status-im/status-react/pull/3605). The same events (see https://github.com/status-im/status-react/blob/develop/src/status_im/utils/mixpanel_events.edn for a list of events) can be sent to Instabug using the logging feature described here https://docs.instabug.com/docs/react-native-logging#section-user-steps.

Like Mixpanel, we should only send these events for users that opt-in.

cc @hesterbruikman @jpbowen 

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-react">status-im/status-react</a></strong></div><div>status-react - a free (libre) open source, mobile OS for Ethereum</div></blockquote>
<blockquote><div><strong><a href="https://docs.instabug.com/docs/react-native-logging#section-user-steps">Logging</a></strong></div><div>You can send a variety of logs types with each crash or bug report. They will appear withing each report on your Instabug dashboard as shown below.   ## Instabug Log  You can log messages throughout your application's lifecycle. Those logs are going to be sent with each report.  ## User Steps  Insta...</div></blockquote>

### PR DESCRIPTION
**Pull request by dmitryn**
fixes #3738 

Send user events to Instabug to allow running custom surveys based on user behavior.

PR also removes custom hacks to manually show survey when N events has happened (this was done as a workaround while Instabug User Events API was not working)

status: ready

### CODE REVIEW

**Comment by t1MZXxBn6pAAtTqKTy5UotdeMqBw2CfGrUF** [comment]
These are very good labels.

**Review by jeluard — APPROVED** [review]

File: src/status_im/utils/handlers.cljs [hunk 1]
```diff
@@ -103,9 +103,8 @@
                anon-id  (ethereum/sha3 current-account-id)]
            (doseq [{:keys [label properties]}
                    (mixpanel/matching-events new-db event mixpanel/event-by-trigger)]
-             (mixpanel/track anon-id label properties offline?))
-           (when (= :send-current-message (first event))
-             (instabug/maybe-show-survey new-db)))))
+             (mixpanel/track anon-id label properties offline?)
+             (instabug/track label properties)))))
```
Comment (rasom) [inline src/status_im/utils/handlers.cljs:105]: We have a queue for mixpanel events (which stores events while app is offline), do we need the same fo instabug?

Comment (dmitryn) [reply src/status_im/utils/handlers.cljs:105]: @rasom not necessary, it's okay to loose some of instabug events as they only used to show surveys

**Review by rasom — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@dmitryn looks like it works fine 👍 

Here are some insights from my testing:
- Events are logged in this format `Tap :wallet-send-transaction {}`
- The same events are shared again when multiple bugs are reported in the same session. When app is killed and opened again, only new events are shared. But, I suppose it's the way it should work
- New events are not shared when data sharing is disabled which is correct
- Mixpanel events tracking works fine
- I checked Android and iOS

@chadyj wdyt about running a test survey based on some tap events? We could do it as a part of Status Everyday.

**Comment by flexsurfer** [comment]
@dmitryn could you rebase onto develop, and `cljfmt check` thanks

**Pull request merged**